libflux/src/flux/semantic/builtins.rs

```diff
@@ -350,7 +350,7 @@ pub fn builtins() -> Builtins<'static> {
                 "splitAfter" => "forall [] (v: string, t: string) -> [string]",
                 "splitN" => "forall [] (v: string, t: string, n: int) -> [string]",
                 "splitAfterN" => "forall [] (v: string, t: string, i: int) -> [string]",
-                "joinStr" => "forall [] (arr: [string], v: string) -> {}",
```

Comment:
Oh I didn't realize that the reason for the pandas failure was just a type failure.

Reply:
Yeah this wasn't caught before because we were looking at the wrong property which had a valid type (integer). We never actually added the invalid property to the map.

Note if we added a new kind constraint asserting that rows returned from any of the transformation functions could only contain basic types, this kind of error could be caught at compile time. Future work.